panic querying CSI plugins

On nightly (built from last night's master):

```
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:     2020-08-13T14:40:42.396Z [ERROR] http: http: panic serving 100.19.30.237:61544: assignment to entry in nil map
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: goroutine 10675 [running]:
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.(*conn).serve.func1(0xc0003caa00)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:1772 +0x139
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: panic(0x2b80ae0, 0x38177f0)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         runtime/panic.go:973 +0x396
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: github.com/hashicorp/nomad/command/agent.structsAllocListStubToApi(0xc0011d8f20, 0xc00100df20)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         github.com/hashicorp/nomad/command/agent/csi_endpoint.go:423 +0x404
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: github.com/hashicorp/nomad/command/agent.structsCSIPluginToApi(0xc000c5b180, 0xc00100ad80)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         github.com/hashicorp/nomad/command/agent/csi_endpoint.go:299 +0x447
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: github.com/hashicorp/nomad/command/agent.(*HTTPServer).CSIPluginSpecificRequest(0xc000326b90, 0x38edee0, 0xc00100ad80, 0xc001328200, 0x4d7086, 0x5f3550ea, 0x179f92e6, 0x758b0becfe)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         github.com/hashicorp/nomad/command/agent/csi_endpoint.go:265 +0x351
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: github.com/hashicorp/nomad/command/agent.(*HTTPServer).wrap.func1(0x38edee0, 0xc00100ad80, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         github.com/hashicorp/nomad/command/agent/http.go:448 +0x176
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.HandlerFunc.ServeHTTP(0xc0006663c0, 0x38edee0, 0xc00100ad80, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:2012 +0x44
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.(*ServeMux).ServeHTTP(0xc00004e600, 0x38edee0, 0xc00100ad80, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:2387 +0x1a5
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1(0x38f7f60, 0xc000204460, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         github.com/NYTimes/gziphandler@v1.0.1/gzip.go:277 +0x1e6
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.HandlerFunc.ServeHTTP(0xc0006b60c0, 0x38f7f60, 0xc000204460, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:2012 +0x44
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.serverHandler.ServeHTTP(0xc0002040e0, 0x38f7f60, 0xc000204460, 0xc001328200)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:2807 +0xa3
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: net/http.(*conn).serve(0xc0003caa00, 0x3909ea0, 0xc001002ac0)
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:1895 +0x86c
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]: created by net/http.(*Server).Serve
Aug 13 14:40:42 ip-172-31-93-242 nomad[13410]:         net/http/server.go:2933 +0x35c
```

--- a/command/agent/csi_endpoint.go
+++ b/command/agent/csi_endpoint.go
@@ -407,6 +407,7 @@ func structsAllocListStubToApi(alloc *structs.AllocListStub) *api.AllocationList
 		DesiredDescription:    alloc.DesiredDescription,
 		ClientStatus:          alloc.ClientStatus,
 		ClientDescription:     alloc.ClientDescription,
+		TaskStates:            make(map[string]*api.TaskState, len(alloc.TaskStates)),
 		FollowupEvalID:        alloc.FollowupEvalID,
 		PreemptedAllocations:  alloc.PreemptedAllocations,
 		PreemptedByAllocation: alloc.PreemptedByAllocation,
@@ -445,7 +446,7 @@ func structsRescheduleTrackerToApi(rt *structs.RescheduleTracker) *api.Reschedul
 	if rt == nil {
 		return nil
 	}
-	out := &api.RescheduleTracker{}
+	out := &api.RescheduleTracker{Events: []*api.RescheduleEvent{}}
 
 	for _, e := range rt.Events {
 		out.Events = append(out.Events, &api.RescheduleEvent{
@@ -470,6 +471,7 @@ func structsTaskStateToApi(ts *structs.TaskState) *api.TaskState {
 		LastRestart: ts.LastRestart,
 		StartedAt:   ts.StartedAt,
 		FinishedAt:  ts.FinishedAt,
+		Events:      []*api.TaskEvent{},
 	}
 
 	for _, te := range ts.Events {

